framework/src/modules/token/commands/transfer.ts

```diff
@@ -43,6 +48,42 @@ export class TransferCommand extends BaseCommand {
 
 		try {
 			validator.validate(transferParamsSchema, params);
+
+			const senderAddress = utils.hash(context.transaction.senderAddress);
+			const tokenID = context.transaction.senderAddress;
+			const userStore = this.stores.get(UserStore);
+
+			const balanceChecks: { [key: string]: bigint } = {};
+
+			if (userStore.getKey(params.recipientAddress, tokenID).length === 0) {
+				if (params.accountInitializationFee !== USER_SUBSTORE_INITIALIZATION_FEE) {
+					throw new Error('Invalid account initialization fee');
+				}
+
+				balanceChecks[
+					TOKEN_ID_STORE_INITIALIZATION.toString('hex')
+				] = USER_SUBSTORE_INITIALIZATION_FEE;
+			}
+
+			const tokenIDSerialized = tokenID.toString('hex');
+
+			if (balanceChecks[tokenIDSerialized] !== undefined) {
+				balanceChecks[tokenIDSerialized] += params.amount;
+			} else {
+				balanceChecks[tokenIDSerialized] = params.amount;
+			}
+
+			for (const tokenIDKey of Object.keys(balanceChecks)) {
+				const availableBalance = await this._method.getAvailableBalance(
+					context.getMethodContext(),
+					senderAddress,
+					Buffer.from(tokenIDKey),
```

Comment:
When you call `Buffer.from` default is `utf-8`, so you need to specify `hex` if you inserted as `hex`